### ISSUE TITLE
Adding single Particles2D node cuts framerate in halve

### ISSUE THREAD

**Issue by golddotasksquestions** [godotengine/godot#25309]
**Godot version:**
Godot 3.0.6, Godot 3.1 beta2

**OS/device including version:**
Win7 Professional i7-4800MQ CPU @ 2.7 GHz
GeForce GTX 765M 2048MB 
https://www.geforce.com/hardware/notebook-gpus/geforce-gtx-765m/specifications
latest drivers installed

**Issue description:**
Adding a simple Particles2D node cuts framerate in halve.
I've simply added one Particles2D node and assigned ParticlesMaterial. Suddenly 60 suddenly drops to 30 FPS when **vsync is on**. This is also the case in 3.1 beta2 if Particles are emiting. In 3.0.6 also if they are not emitting.

Godot has been assigned to always run on the dedicated graphics card inside the Nvidia driver. There is a led-indicator light for that purpose that glows when I start Godot.

Turning off Nvidia vsync and **one particle node**: 60-50 FPS + terrible frametime jitter when camera is moving (also unplayable)
Turning off Nvidia vsync and **no particle node**: 120-100 FPS + terrible frametime jitter when camera is moving (also unplayable)

-> Notice again: **halve the framerate**

Turning off Nvidia vsync and Godot vsync off and **one particle node**: 120-100 FPS + terrible frametime jitter when camera is moving (also unplayable)

Strange behavior when first turning vsync off in Nvidia as well as Godot and then back on while Godot 3.0.6 is still running: Solid 60 FPS, plus no frametime jitter at all. Perfect! ... 
 ... however only until I restart Godot  ... then it's back to 30 FPS and therefore visible lag. (back to unplayable)

**Comment by Calinou** [comment]
Out of curiosity, does this improve if you disable Windows Aero?

**Comment by golddotasksquestions** [comment]
Not an improvement, but it does make a significant difference: With Windows Aero disabeled but vsync on in the Nvidia Driver as well as in Godot plus one standard Particle2D node, I get solid 60 FPS instead of the 30, but the frametime stutter is also clearly here. Which is at least as irritating as the low framerate. It looks like vsync would be disabled, even though both settings say it's on, when Areo is disabled. Turning Aero back on and restarting Godot removes the stutter but again, halve the FPS because of one Particle2D.

**Comment by golddotasksquestions** [comment]
I've just uninstalled all Nvidia drivers. So I would assume Godot is running just on the internal Intel graphics. The problem exactly the same however.

**Comment by TheDuriel** [comment]
While i can't replicate the exact scenario, having a particle emitter with a material visible the in scene (2D) does in fact have a large impact on framerate. (Unlocked) and introduces spikes. Irregardless of if anything is being drawn.

https://cdn.discordapp.com/attachments/540803169374961664/565304301019463681/unknown.png

Off, On, Off (Ingore the spikes in the second Off, i alt+tabbed.)

I imagine that this expected? drop is may be amplified by OPs specific work environment.

**Comment by Calinou** [comment]
@golddotasksquestions (or anyone else) Can you reproduce this on [Godot 3.2 beta4](https://godotengine.org/article/dev-snapshot-godot-3-2-beta-4)

As for stuttering when using V-Sync, [this merged pull request](https://github.com/godotengine/godot/pull/33414) might help. You'll have to enable **Display > Window > Vsync > Vsync Via Compositor** in the Project Settings for it to be effective.

**Comment by golddotasksquestions** [comment]
Thank you @Calinou for getting back to this issue! 
I tested it again now in 3.1.2 stable as well as 3.2. beta 4.
In 3.1.2 it is present just as described in the original post, in 3.2.beta4 it is not present any more. No Framerate drop at all in 3.2. beta4 !! :)
(I noticed some other weird particle bug in 3.2.beta4 while testing this though: https://github.com/godotengine/godot/issues/34713)
